File: src/main/java/one/theaq/handsmanager/tag/CommonTags.java
package one.theaq.handsmanager.tag;

import net.minecraft.core.registries.Registries;
import net.minecraft.tags.TagKey;
import net.minecraft.world.item.Item;
import one.theaq.handsmanager.HandsManagerMain;

public class CommonTags {
	public static TagKey<Item> CROSSBOWS = TagKey.create(Registries.ITEM, HandsManagerMain.INSTANCE.commonLocation("tools/crossbow"));
	
	public static void init() {}
}

File: src/main/kotlin/one/theaq/handsmanager/tag/CommonTags.kt
package one.theaq.handsmanager.tag

import net.minecraft.core.Registry
import net.minecraft.core.registries.Registries
import net.minecraft.resources.ResourceKey
import net.minecraft.tags.TagKey
import one.theaq.handsmanager.HandsManagerMain.commonLocation

object CommonTags {
    var CROSSBOWS = createCommonTag(Registries.ITEM, "tools/crossbow")

    fun <registryType : Any> createCommonTag(
        registry: ResourceKey<Registry<registryType>>,
        location: String
    ): TagKey<registryType> {
        return TagKey.create(registry, commonLocation(location))
    }
}